### DISCUSSION TITLE
IP | System Design

### DISCUSSION THREAD

**Discussion by giscus** [karanpratapsingh/portfolio#36] [Comments]
# IP | System Design

In this tutorial, we will discuss IP

https://www.karanpratapsingh.com/courses/system-design/ip

**Comment by Aztinct** [comment]
The IP is the Internal Protocol/Rules of communication. This is the set of rules that data sent over the internet or local network all follow. IP Address are like a physical address, they let you know where is the information coming from. This helps the internet to know whether is it a computer, router, or website. 

There are different IPs. IPv4 was what we started with but we ran out of addresses soon. Now we're on IPv6 and we have enough addresses for awhile. 

Public IP address refers to having 1 address being shared by all the connected devices. 
Private IP address is the unique IP number assigned to every device. E.g. Your router generates a unique IP for your phone, then another for your PC that is connected to it. With this address, information you requested from another IP know where to send those information to. 
Static IP address doesn't change but are more expensive. They are used for impt things such as reliable geo-location, remote access, hosting servers, etc. 
Dynamic IP address changes. It's assigned by the Dynamic Host Configuration Protocol. 
They are the most common IP protocol addresses. Cheaper to deploy and reuse IP addresses within network as needed. 

IP addresses can vary from shared public ones to private IPs differentiating each device connected to the router. Static is useful for contexts which requires high realibility and low variance. With an unchanging IP, miscommunication and errors are less likely. Dynamic IPs can be reused.
